### PR TITLE
Persist DHT node datastore and restore contacts on startup

### PR DESCRIPTION
**Pull request by jackrobison**

### CODE REVIEW

File: lbrynet/dht/node.py [hunk 1]
```diff
@@ -51,7 +52,7 @@ class Node(object):
 
     def __init__(self, node_id=None, udpPort=4000, dataStore=None,
                  routingTableClass=None, networkProtocol=None,
-                 externalIP=None):
+                 externalIP=None, dataDirectory=None, fileName=None):
         """
         @param dataStore: The data store to use. This must be class inheriting
                           from the C{DataStore} interface (or providing the
```
Comment (kaykurokawa) [inline lbrynet/dht/node.py]: need better docstring for dataDirectory (describe what it does when None, etc..)  , need docstring for fileName

File: lbrynet/dht/datastore.py [hunk 2]
```diff
@@ -13,11 +22,22 @@ def __init__(self):
         # Dictionary format:
         # { <key>: (<value>, <lastPublished>, <originallyPublished> <originalPublisherID>) }
         self._dict = {}
+        self._node_state = {}
+
```
Comment (kaykurokawa) [inline lbrynet/dht/datastore.py]: add docstring for self._node_state

File: lbrynet/dht/datastore.py [hunk 3]
```diff
@@ -46,7 +66,64 @@ def getPeersForBlob(self, key):
...
+    def Save(self, contacts):
+        pass
+
+
+class JSONFileDataStore(DictDataStore):
+    def __init__(self, node, path, file_name=None):
+        DictDataStore.__init__(self)
+        self._node = node
+        self._path = os.path.join(path, file_name or "dht_datastore.json")
```
Comment (kaykurokawa) [inline lbrynet/dht/datastore.py]: Doesn't look like JSONFileDataStore uses self._node
If it does, you have a circular dependency

**Comment by tzarebczan** [comment]
@jackrobison will this PR be revived or can it be closed?

**Pull request closed without merging**